sdk/search/Azure.Search.Documents/src/Indexes/Models/AzureMachineLearningSkill.cs

```diff
@@ -0,0 +1,319 @@
+        /// <param name="authenticationKey">The key for the AML service.</param>
+        /// <param name="scoringUri">The scoring URI of the AML service to which the JSON payload will be sent. Only the https URI scheme is allowed.</param>
+        public AzureMachineLearningSkill(IEnumerable<InputFieldMappingEntry> inputs, IEnumerable<OutputFieldMappingEntry> outputs, Uri scoringUri, string authenticationKey) :
+            this(inputs, outputs, scoringUri)
+        {
+            AuthenticationKey = authenticationKey ?? throw new ArgumentNullException(nameof(authenticationKey));
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureMachineLearningSkill"/> class.
+        /// </summary>
+        /// <param name="inputs">Inputs of the skills could be a column in the source data set, or the output of an upstream skill.</param>
+        /// <param name="outputs">The output of a skill is either a field in a search index, or a value that can be consumed as an input by another skill.</param>
+        /// <param name="resourceId">The Azure Resource Manager resource Id of the Aml service. It should be in the format subscriptions/{guid}/resourceGroups/{resource-group-name}/Microsoft.MachineLearningServices/workspaces/{workspace-name}/services/{service_name}.</param>
+        public AzureMachineLearningSkill(IEnumerable<InputFieldMappingEntry> inputs, IEnumerable<OutputFieldMappingEntry> outputs, string resourceId) :
+            this(inputs, outputs)
+        {
+            if (resourceId == null)
+            {
+                throw new ArgumentNullException(nameof(resourceId));
+            }
+
+            ResourceId = new ResourceIdentifier(resourceId);
+        }
+
+        /// <summary> Initializes a new instance of AmlSkill. </summary>
+        /// <param name="inputs"> Inputs of the skills could be a column in the source data set, or the output of an upstream skill. </param>
+        /// <param name="outputs"> The output of a skill is either a field in a search index, or a value that can be consumed as an input by another skill. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="inputs"/> or <paramref name="outputs"/> is null. </exception>
+        internal AzureMachineLearningSkill(IEnumerable<InputFieldMappingEntry> inputs, IEnumerable<OutputFieldMappingEntry> outputs) : base(inputs, outputs)
+        {
+            if (inputs == null)
+            {
+                throw new ArgumentNullException(nameof(inputs));
+            }
+            if (outputs == null)
+            {
+                throw new ArgumentNullException(nameof(outputs));
+            }
+
+            ODataType = "#Microsoft.Skills.Custom.AmlSkill";
+        }
+
+        void global::Azure.Core.IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
```

Comment:
Why do we need to do all this?

Reply:
CodeGen did not generate compilable code for `ResourceIdentifier` and `AzureLocation`.
I have opened [an issue](https://github.com/Azure/autorest.csharp/issues/1933).

Reply:
Oh, I see.  In that case, can we do the "use an internal property as a makeshift converter" trick so we don't have to own updating the serialization code whenever the swagger changes?  Imagine something like:

```C#
        [CodeGenMember("Region")]
        internal string RawLocation
        {
            get => Location?.ToString();
            set => Location = (value == null ? null : new AzureLocation(value));
        }
        
        public AzureLocation? Location { get; private set; }
```